src/helpers.js

```diff
@@ -80,6 +84,109 @@ function paramsForAction(action) {
   return results;
 }
 
+// Removes line breaks but preserves paragraphs (double newlines).
+// Also reserves line breaks denoted with an optional delimiter.
+// TODO: make this un-hacky
+function removeLineBreaks(text, opt_paragraph_delim) {
+  var paragraph_delim = opt_paragraph_delim || "\n\n";
+  text = text.replace(/\\\n/gm, "XX");
+  text = text.replace(/\n\n/gm, "CC");
+  text = text.replace(/\r\n|\n|\r/gm, " ");
+  text = text.replace(/XX/g, "\n");
+  return text.replace(/CC/g, paragraph_delim);
+}
+
+function genericPath(action, pathParams) {
+  var path = action.path;
+  _.forEach(pathParams, function(pathParam) {
+    path = path.replace(/%./, pathParam.name + "-id");
+  });
+  return path;
+}
+
+function samplePath(action, pathParams) {
+  var path = action.path;
+  _.forEach(pathParams, function(pathParam) {
+    path = path.replace(/%./, pathParam.example_values[0]);
+  });
+  return path;
+}
+
+function addDataForSpecialActions(resource, action, data) {
+  if (resource === 'task') {
+    switch(action) {
+      case 'addSubtask':
+        // We don't have templates for subtasks but would like the copy to be different from regular tasks
+        data.push('-d "name=\'Make trip to Cats R Us\'"');
```

Comment:
So here, I would think that the `data` would actually be a key-value map rather than an array? Because the `-d "..."` syntax is specific to curl. If we keep it in this format, it's too specific to curl and couldn't be used to generate, say, example JS client code.